docs/enhancements/cancel-day-2-host-installation.md

```diff
@@ -0,0 +1,177 @@
+---
+title: cancel-day-2-host-installation
+authors:
+  - "@CrystalChun"
+creation-date: 2025-04-22
+last-updated: 2025-04-22
+---
+
+# Cancel Day-2 Host Installation
+
+## Summary
+
+In Assisted-Service's Kubernetes API implementation, there is currently no way to cancel the
+installation of a day-2 host. This leads to scenarios where an Agent can become stuck for a long
+time if the installation fails and the timeout hasn't been reached, or if the user changes their 
+mind about installing a host before the the installation finishes. This proposal aims to allow
+cancellation of installation for a day-2 host.
+
+## Motivation
+
+There would be times a user is installing a day-2 host and in the middle of installation, they need to
+change something, or the install becomes stuck because of an error. There are also instances where a
+user might need to provision and deprovision a host quickly as it may no longer be needed, such as when
+autoscaler is used. 
+
+When a host starts installation, there's no way to remove it from a cluster until the host installation
+times out and errors out or it finishes installation. These can take a very long time, and it can be a
+frustrating experiences for users.
+
+### Goals
+
+- Once a day-2 host starts installation, the installation can be cancelled
+- There's a straight-forward way to trigger cancellation
+- Host whose installations are cancelled can be reclaimed back to their InfraEnvs and be reused
+- Host cancellation should not affect other hosts in the cluster and should not affect the cluster itself
+- Host cancellation should happen within a reasonable amount of time in order to support dynamic scaling 
+
+### Non-Goals
+
+- Hosts that are installed in a day-1 cluster is out of scope of this enhancement
+- Cluster installation cancellation is out of scope
+- The way reclaim/unbind works should not be affected by these changes
+- This enhancement is focused solely on Assisted-Service offerings using Kubernetes API (SaaS should not be affected by this)
+- The host should still be responsive, in other words, any hosts that are unreachable are out scope of this proposal
+
+## Proposal
+
+Allow cancelling an installing day-2 host through Assisted-Service's Kubernetes API.
+
+### User Stories
+
+#### Story 1
+
```

Comment:
May be good to make it clear in the introduction, or even in the title, that this is for Hypershift only.

Reply:
Will do, thank you!